### PR TITLE
[WIP]Add logs to alert-details page

### PR DESCRIPTION
**Pull request by sangupta1**
This PR restructures the `Alert details` page to have logs for each `Alert`.
Upon clicking a particular `Alert`, that `Alert` opens up. It would contain a horizontal Navbar with 2 navbar options- `Details` and `Logs`.
The `Details` option will show the details for the respective `alert`. 
The `Logs` option will show logs for the specific `Pod` and `Namespace` which is affected by the selected `Alert`. These logs are fetched from the `Log Exploration Api` filtered upon a specific time duration and number of log lines referred to as `Max Logs`. The time duration and the `Max Logs` are provided as inputs which the user can provide to get logs filtered as per these parameters. If an Alert does not have a `Pod` and `Namespace` associated with it, then the `Logs` option will show a message `No alert logs found`. If Logging is not installed or the `Log Exploration Api` is not accessible, then only the `Details` navbar option is shown and the `Logs` navbar option is not shown.
JIRA issue- https://issues.redhat.com/browse/LOG-1192

The screenshots for the changes this PR is making are- 
1.) The `Alert details` Page having 2 navbar options- `Details` and `Logs`. 
Screenshot from 2021-05-17 19-51-18

2.) The Logs displayed by going to the `Logs` option. By default, 1000 logs from the specific `Pod` affected by the Alert are shown. These logs are returned by the `Log Exploration Api`. (In the screenshot is the `Update Available` alert which has a `Pod` and `namespace` affected by this alert. 
Screenshot from 2021-06-03 16-34-58

3.) In this screenshot the `Update Available` alert is shown. Here the time duration selected is 0 hours and 30 minutes. It means the logs will be fetched for the duration of 30 minutes before the `Alert` is activated. The `Max Logs` entered are 7 which means 7 log lines will be fetched from the `Log Exploration Api`.
Screenshot from 2021-06-03 17-10-50

4.) In this screenshot the `AlertmanagerReceiversNotConfigured` Alert is shown. It does not have a `Pod` and `Namespace` linked to it. (`Watchdog` is another such alert). Hence on moving to the `Logs` tab for this alert , a message `No alert logs found` is shown.
Screenshot from 2021-05-17 19-51-46
Screenshot from 2021-05-20 16-27-38

The gif for the Changes for the Administrator perspective-
Changes_for_administrator_perspective

The gif for the Changes for the Developer perspective-
Changes_for_Developer_perspective

The steps for configuring the Log Exploration API are in the attached doc file-
https://docs.google.com/document/d/110I_pEMDL6DljdrZmSFMsneHn4ZmnBKXT_O2juHJrjg/edit?usp=sharing

### CODE REVIEW

**Comment by divyanshiGupta** [comment]
@sangupta1 re-title your PR with a more appropriate title.

**Comment by sangupta1** [comment]
`/retest`

**Comment by divyanshiGupta** [comment]
@sangupta1 it seems like there is a build error. Run `build.sh` and see what the error is and try to fix it. Also make sure you have the latest dependencies and your code is compiled without any errors.

File: frontend/public/components/monitoring/alert-logs-utils.ts
Comment (divyanshiGupta) [inline frontend/public/components/monitoring/alert-logs-utils.ts:71]: Here you are importing from alert-logs file and in alert-logs file you are importing from here so it might create cyclic dependency. Move this function to alert-logs file.

File: frontend/public/components/monitoring/alert-logs.tsx
Comment (divyanshiGupta) [inline frontend/public/components/monitoring/alert-logs.tsx:21]:
```suggestion
const AlertLogs: React.FC<AlertLogsProps> = (props) => {
```

File: frontend/public/components/monitoring/alert-logs.tsx
Comment (divyanshiGupta) [inline frontend/public/components/monitoring/alert-logs.tsx:71]:
```suggestion
       <EmptyBox label="alert logs" />
```

**Review by divyanshiGupta — CHANGES_REQUESTED** [review]

File: frontend/public/components/monitoring/alert-logs.tsx
Comment (divyanshiGupta) [reply frontend/public/components/monitoring/alert-logs.tsx:81]: Also move these type definitions above the component.

**Comment by divyanshiGupta** [comment]
@sangupta1 looks like you need to run `yarn i18n`.

File: frontend/public/components/monitoring/alerting.tsx
Comment (divyanshiGupta) [reply frontend/public/components/monitoring/alerting.tsx:372]: And why have you converted these components into functions? I dont think so there is a need to do that.

File: frontend/public/components/monitoring/alerting.tsx
Comment (divyanshiGupta) [inline frontend/public/components/monitoring/alerting.tsx:402]: Same as above

File: frontend/public/components/monitoring/alerting.tsx
Comment (divyanshiGupta) [inline frontend/public/components/monitoring/alerting.tsx:422]: Same as above

File: frontend/public/components/monitoring/alerting.tsx
Comment (divyanshiGupta) [reply frontend/public/components/monitoring/alerting.tsx:372]: If you did this just to export them then you export the component itself.

**Review by divyanshiGupta — CHANGES_REQUESTED** [review]

File: frontend/public/components/monitoring/alerting.tsx
Comment (sangupta1) [reply frontend/public/components/monitoring/alerting.tsx:422]: @divyanshiGupta I did not convert them into functions. I am sorry. I think during rebase with console/master and resolving merge conflict, i did this mistake and accepted an older version. I will rebase and fix this.

File: frontend/public/components/monitoring/alerting.tsx
Comment (divyanshiGupta) [inline frontend/public/components/monitoring/alerting.tsx:750]:
```suggestion
            // t('public~Logs')
```

File: frontend/public/components/monitoring/alerting.tsx
Comment (divyanshiGupta) [inline frontend/public/components/monitoring/alerting.tsx:737]:
```suggestion
            // t('public~Details')
```

**Review by divyanshiGupta — CHANGES_REQUESTED** [review]

File: frontend/public/locales/en/details-page.json
Comment (divyanshiGupta) [inline frontend/public/locales/en/details-page.json:4]: It seems it's not used anywhere now. Run yarn i18n and that should remove these strings. Also remove this file.

**Review by divyanshiGupta — CHANGES_REQUESTED** [review]

**Comment by divyanshiGupta** [comment]
@sangupta1 update your screenshots with the current changes. And if possible add a gif as well.

**Comment by sangupta1** [comment]
@divyanshiGupta I added the relevant gifs for the changes.

**Comment by divyanshiGupta** [comment]
Note: The feature being added by this PR is targeted for 4.9.

**Comment by divyanshiGupta** [comment]
/kind feature

Note: The feature being added by this PR is targeted for 4.9.

**Comment by divyanshiGupta** [comment]
@Sanyam Gupta we can keep your PR in WIP and add the Log exploration api integration changes in this PR only. Also make sure to add all the setup steps required for accessing the API in PR description and the necessary gifs/screenshots.

**Comment by sangupta1** [comment]
@divyanshiGupta I have updated the PR with the latest changes and added all the steps for the configuration of the API as well. Also updated all the GIF's. One of the tests is failing, i am working towards fixing it.

**Comment by divyanshiGupta** [comment]
@sangupta1 also add the link for the backend service this PR is blocked by and also mention that after merging the backend PR this PR can be merged.

**Pull request closed without merging**
